src/modules/rover_ackermann/RoverAckermann.cpp

```diff
@@ -110,20 +94,42 @@ void RoverAckermann::Run()
 
 }
 
-void RoverAckermann::manualControl()
+void RoverAckermann::generateSetpoints()
 {
-	if (_vehicle_control_mode.flag_control_position_enabled) {
-		_manual_mode.position();
+	vehicle_status_s vehicle_status{};
+	_vehicle_status_sub.update(&vehicle_status);
 
-	} else if (_vehicle_control_mode.flag_control_attitude_enabled) {
-		_manual_mode.stab();
+	switch (vehicle_status.nav_state) {
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION:
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER:
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_RTL:
+		_auto_mode.autoControl();
```

Comment:
How is the controller architecure again: an external mode is never considered "auto" and thus `_auto_mode` is never called when in an external mode?